graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java

```diff
@@ -109,8 +109,8 @@ protected HttpResponse doPost(ServiceRequestContext ctx, HttpRequest request) th
                         return HttpResponse.of(HttpStatus.BAD_REQUEST, MediaType.PLAIN_TEXT, "Missing query");
                     }
                     final String operationName = (String) requestMap.get("operationName");
-                    final Map<String, Object> variables = toMap(requestMap.get("variables"));
-                    final Map<String, Object> extensions = toMap(requestMap.get("extensions"));
+                    final Map<String, Object> variables = toMapFromJson(requestMap.get("variables"));
```

Comment:
Note: I feel like it might be worth `try..catch`ing `toMapFromJson` and returning a `BAD_REQUEST`.

As an example, if IAE is thrown the response code will end up being 500.

https://github.com/line/armeria/blob/7e11984205e4325f510b522f81fb241e13ce21da/graphql-protocol/src/main/java/com/linecorp/armeria/server/graphql/protocol/AbstractGraphqlService.java#L175

Having said this, I don't think this needs to be handled in this PR

Reply:
I agree to both statements. 500 is wrong when it's clearly a client error, and that it shouldn't be changed in this PR.